[Silabs] [Wi-Fi] [SiWx917] Added support for new SDK 3.1.1 for 917 NCP

shgutte opened the pull request:
Description of Problem/Feature:
Added support for new SDK 917 NCP as well  as  gsdk 4.4.0 

Description of Fix/Solution:
Added support for new SDK 917 NCP as well  as  gsdk 4.4.0 for for board 4186C and 4197C 

Testing Done:
917 NCP + 4186C
917 NCP + 4187C

shgutte committed:
Added changes for new wifi sdk 917 NCP

shgutte committed:
Added changes for 917 NCP

shgutte committed:
Added changes for rs9117 NCP

shgutte committed:
Added changes sl_eifi_if file

shgutte committed:
Added changes for 917 commissioning

shgutte committed:
Updated matter support

shgutte committed:
Added chnages for power manager files

shgutte committed:
Updated matter support

shgutte committed:
Added changes for 917 NCP bringup

shgutte committed:
Removed the changes for 917 NCP

shgutte committed:
Added changes 917 NCP bringup

shgutte committed:
Updated matter support

jmartinez-silabs commented on third_party/silabs/efr32_sdk.gni:
```diff
@@ -130,6 +130,7 @@ template("efr32_sdk") {
     # Treat these includes as system includes, so warnings in them are not fatal.
     _include_dirs = [
       "${efr32_sdk_root}",
+      "${examples_plat_dir}",
```
This is strange and is not part of the efr32_sdk. what are we including this here?

shgutte replied:
we are trying to include spi_multiplex.h from that location

jmartinez-silabs replied:
Maybe spi_multiplex.h and related code should not be in `examples_plat_dir` if it is needed by SDK files.

the dependency should be the other way around (examples_plat_dir sources requiring the SDK. not SDK requiring examples_plat_dir)

Please add a TODO comment explain why it is included three and that it need to be fixed/cleaned

shgutte replied:
Added TODO

jmartinez-silabs commented on src/platform/silabs/rs911x/BLEManagerImpl.cpp:
```diff
@@ -78,15 +84,26 @@ using namespace ::chip::DeviceLayer::Internal;
+#if SIWX_917
+    sl_status_t sl_status;
+    //! Get Random number of desired length
+    sl_status = sl_si91x_trng_get_random_num((uint32_t *)randomAddrBLE, RSI_BLE_ADDR_LENGTH);
```
this work is already done in another PR isn't it?  It is a bit confusing to have the same additions in 2 pr

shgutte replied:
removed

shgutte replied:
rebased the branch

jmartinez-silabs commented on examples/platform/silabs/efr32/rs911x/wfx_rsi_host.c:
```diff
@@ -31,10 +31,10 @@
 #include "wfx_host_events.h"
 #include "wfx_rsi.h"
 
+
```
No changes in this file really should probably be left as it was

shgutte replied:
it's same file i don't know why it's showing + line

jmartinez-silabs commented on examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.c:
```diff
@@ -301,6 +364,25 @@ static sl_status_t wfx_rsi_init(void)
         return status;
     }
+#ifdef SIWX_917
+    const uint32_t trngKey[TRNGKEY_SIZE] = { 0x16157E2B, 0xA6D2AE28, 0x8815F7AB, 0x3C4FCF09 };
```
same comment: this code is added in another PR that I reviewed and is not needed to enable this PR. You should not be replicating this work again.

shgutte replied:
rebased the branch

shgutte replied:
rebased the branch

jmartinez-silabs commented on examples/platform/silabs/efr32/rs911x/hal/efx32_ncp_host.c:
```diff
@@ -0,0 +1,287 @@
+/*******************************************************************************
```
isn't that file in wifi_sdk ?

shgutte replied:
Yes we have updated it for SPI multiplexing changes so copied till they add the API for SPI multiplexing

jmartinez-silabs commented on examples/platform/silabs/efr32/rs911x/hal/sl_si91x_ncp_utility.c:
```diff
@@ -0,0 +1,282 @@
+ * @return
+ *        None
+ **********************************************************/
+sl_status_t sl_wfx_host_pre_lcd_spi_transfer(void)
```
isn't the majority of this code already in efx_spi.c?

shgutte replied:
Yeah we will create this so both efx_spi as well as efx32_ncp_host files can you have this utility function going further soon we will remove it from efx_spi file and use this file for definitions

chirag-silabs commented on examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.c:
```diff
@@ -41,6 +41,16 @@
 
 #include "ble_config.h"
 
+#if SL_ICD_ENABLED && SIWX_917
```
These changes were already added as part of my PR. Rebase the PR these should be removed

shgutte replied:
rebased

shgutte replied:
rebased the brach

chirag-silabs commented on examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.c:
```diff
@@ -198,6 +213,44 @@ sl_status_t join_callback_handler(sl_wifi_event_t event, char * result, uint32_t
 
 #if SL_ICD_ENABLED
+
+#if SIWX_917
```
Same here

shgutte replied:
rebased

chirag-silabs commented on third_party/silabs/efr32_sdk.gni:
```diff
@@ -204,11 +205,21 @@ template("efr32_sdk") {
-        "${wifi_sdk_root}/components/si91x/inc",
         "${wifi_sdk_root}/components/si91x/memory",
-        "${wifi_sdk_root}/components/si91x/sl_net/inc",
+        "${wifi_sdk_root}/components/device/silabs/si91x/wireless/inc",
```
these should be added in rs9117.gni file right
There is no point of adding this in multiple files

shgutte replied:
Added files in efr32_sdk.gni only if you add it int rs9117.gni it's giving build error for some include files

chirag-silabs commented on src/platform/silabs/rs911x/rsi_ble_config.h:
```diff
@@ -35,12 +36,21 @@
 #define RSI_FAILURE -1
 #endif
 
+#define WIFI_ENABLE_SECURITY_WPA3_TRANSITION 0
```
Why we are defining this to 0 directly

chirag-silabs commented on src/platform/silabs/rs911x/rsi_ble_config.h:
```diff
@@ -35,12 +36,21 @@
 
+#define WIFI_ENABLE_SECURITY_WPA3_TRANSITION 0
+
+#ifndef CHIP_DEVICE_CONFIG_ENABLE_IPV4
```
here as well

shgutte replied:
removed

shgutte committed:
Merge branch 'master' of https://github.com/project-chip/connectedhomeip into bringup/917_ncp_with_new_sdk

shgutte committed:
Added changes for the power manager file

shgutte commented on third_party/silabs/efr32_sdk.gni:
```diff
@@ -631,7 +652,7 @@ template("efr32_sdk") {
       "${efr32_sdk_root}/platform/peripheral/src/peripheral_sysrtc.c",
       "${efr32_sdk_root}/platform/radio/rail_lib/plugin/pa-conversions/pa_conversions_efr32.c",
-      "${efr32_sdk_root}/platform/radio/rail_lib/plugin/rail_util_power_manager_init/sl_rail_util_power_manager_init.c",
+   #   "${efr32_sdk_root}/platform/radio/rail_lib/plugin/rail_util_power_manager_init/sl_rail_util_power_manager_init.c",
```
Uncommented the file

shgutte commented on third_party/silabs/efr32_sdk.gni:
```diff
@@ -204,11 +205,21 @@ template("efr32_sdk") {
       "${sl_ot_efr32_root}",
     ]
 
+    if (silabs_family == "efr32mg24") {
```
Yes but it's not taking the included files for board 4186C we tried a lot of times, and if we separate it it works fine otherwise we are getting build error for this files if you want i can show you the issue

jmartinez-silabs replied:
Sounds odd. There is something else at play here.

shgutte replied:
I'll raise JIRA on this to investigate further

shgutte committed:
Added changes for host file

chirag-silabs commented on examples/platform/silabs/efr32/rs911x/sl_wlan_config.h:
```diff
@@ -34,7 +34,7 @@ static const sl_wifi_device_configuration_t config = {
     .boot_config = { .oper_mode = SL_SI91X_CLIENT_MODE,
```
this file can be used directly right examples/platform/silabs/SiWx917/SiWx917/sl_wlan_config.h?

shgutte replied:
Going further may configuration so we are keeping it separate right now

shgutte committed:
Added changes accroding to comments

jmartinez-silabs commented on examples/platform/silabs/efr32/BUILD.gn:
```diff
@@ -281,7 +281,9 @@ source_set("efr32-common") {
 
   if (chip_enable_wifi) {
-    if (use_rs9116 || use_SiWx917) {
+
+
```
```suggestion
```

jmartinez-silabs commented on examples/platform/silabs/efr32/rs911x/wfx_rsi_host.c:
```diff
@@ -31,10 +31,10 @@
 #include "wfx_host_events.h"
 #include "wfx_rsi.h"
 
+
```
```suggestion
```

jmartinez-silabs commented on src/platform/silabs/efr32/wifi/wfx_notify.cpp:
```diff
@@ -43,6 +43,7 @@
 #include <app/server/Dnssd.h>
 #include <app/server/Server.h>
 
+
```
```suggestion
```

jmartinez-silabs commented on third_party/silabs/efr32_sdk.gni:
```diff
@@ -223,9 +236,13 @@ template("efr32_sdk") {
 
         # network_manager component
         "${wifi_sdk_root}/components/service/network_manager/inc",
+
```
```suggestion
```

jmartinez-silabs commented on third_party/silabs/efr32_sdk.gni:
```diff
@@ -602,6 +626,7 @@ template("efr32_sdk") {
 
   source_set(sdk_target_name) {
     sources = [
+
```
```suggestion
```

shgutte committed:
Updated matter support repo

shgutte committed:
Added TODO for the include files

shgutte committed:
Update examples/platform/silabs/efr32/BUILD.gn

Co-authored-by: Junior Martinez <67972863+jmartinez-silabs@users.noreply.github.com>

shgutte committed:
Update third_party/silabs/efr32_sdk.gni

Co-authored-by: Junior Martinez <67972863+jmartinez-silabs@users.noreply.github.com>

shgutte committed:
Update examples/platform/silabs/efr32/rs911x/wfx_rsi_host.c

Co-authored-by: Junior Martinez <67972863+jmartinez-silabs@users.noreply.github.com>

shgutte committed:
Update src/platform/silabs/efr32/wifi/wfx_notify.cpp

Co-authored-by: Junior Martinez <67972863+jmartinez-silabs@users.noreply.github.com>

shgutte committed:
Added restyle changes

shgutte committed:
Removed unwanted line

shgutte committed:
Added restyler changes

shgutte committed:
Added fix for wf200 build

chirag-silabs committed:
Pull request #1476: MATTER-2944 : Adding the is_debug=false flag

Merge in WMN_TOOLS/matter from bugfix/adding_is_debug_flag to RC_2.3.0-1.3

Squashed commit of the following:

commit 31d39faeff5c49db04a475e1fa0b79d0b73a7456
Author: chbansal <chirag.bansal@silabs.com>
Date:   Thu Jan 11 22:00:30 2024 +0530

    adding is_debug=false by default

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/project-chip/connectedhomeip?pullRequest=31325) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/project-chip/connectedhomeip?pullRequest=31325) before we can accept your contribution.<br/>**10** out of **13** committers have signed the CLA.<br/><br/>:white_check_mark: su-shanka<br/>:white_check_mark: jmartinez-silabs<br/>:white_check_mark: jepenven-silabs<br/>:white_check_mark: Sarthak-Shaha<br/>:white_check_mark: KishorSilabs<br/>:white_check_mark: CuRahman<br/>:white_check_mark: riwaghe<br/>:white_check_mark: chirag-silabs<br/>:white_check_mark: rcasallas-silabs<br/>:white_check_mark: shgutte<br/>:x: rerasool<br/>:x: satyanaag-silabs<br/>:x: stefan-silabs<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/project-chip/connectedhomeip?pullRequest=31325) it.</sub>

shgutte committed:
Merge branch 'master' of https://github.com/project-chip/connectedhomeip into bringup/917_ncp_with_new_sdk

shgutte committed:
Added change to fix the sleep
